experimental/proof-tree/proof-tree.metta

```diff
@@ -555,9 +594,9 @@
   (flatten (: ((plus (: 1 Number)) (: 2 Number)) Number))
   (: (plus (: 1 Number) (: 2 Number)) Number))
 !(assertEqual
-  (flatten (: ((ModusPonens (: bc (→ B C)))
+  (flatten (: ((ModusPonens (: Bc (→ B C)))
                (: ((ModusPonens (: ab (→ A B))) (: a A)) B)) C))
-  (: (ModusPonens (: bc (→ B C)) (: (ModusPonens (: ab (→ A B)) (: a A)) B)) C))
+  (: (ModusPonens (: Bc (→ B C)) (: (ModusPonens (: ab (→ A B)) (: a A)) B)) C))
```

Comment:
For some miraculous reason, in my version of MeTTa (0.2.6+g86cc8296), your code works fine even using `bc` instead of `Bc`.  I think it makes sense because the type checker is supposed to support non-deterministic types.  Could you update your version of MeTTa to the latest, try again, and if it works replace `Bc` by `bc`?